app/src/main/java/org/p2p/wallet/receive/list/TokenListPresenter.kt

```diff
@@ -50,7 +51,8 @@ class TokenListPresenter(
         launch {
             interactor.getTokenListFlow().collect { data ->
                 if (data.result.isNotEmpty()) {
-                    view?.showItems(data.result, scrollToUp)
+                    val filteredResult = data.result.filter { it.mintAddress != Constants.WRAPPED_SOL_MINT }
```

Comment:
You can put filter on flow, maybe it will optimise things a bit

Reply:
this flow is reused in new tokens list